Source/WebCore/Modules/indexeddb/IDBActiveDOMObjectInlines.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <WebCore/ContextDestructionObserverInlines.h>
+#include <WebCore/IDBActiveDOMObject.h>
+#include <WebCore/ScriptExecutionContextInlines.h>
+
+namespace WebCore {
+
+template<typename T, typename... Parameters, typename... Arguments>
+void IDBActiveDOMObject::performCallbackOnOriginThread(T& object, void (T::*method)(Parameters...), Arguments&&... arguments)
+{
+    ASSERT(&originThread() == &object.originThread());
+
+    if (canCurrentThreadAccessThreadLocalData(object.originThread())) {
+        (object.*method)(std::forward<Arguments>(arguments)...);
+        return;
+    }
+
+    Locker<Lock> lock(m_scriptExecutionContextLock);
+
+    ScriptExecutionContext* context = scriptExecutionContext();
+    if (!context)
+        return;
+
+    context->postCrossThreadTask(object, method, arguments...);
+}
+
+inline void IDBActiveDOMObject::callFunctionOnOriginThread(Function<void()>&& function)
```

Comment:
Seems like this function probably just shouldn’t be inlined.

Reply:
> This class was originally a single-header-only one, so no .cpp file exists for this method to be placed in.

Understood. In future I’m sure it would be OK to move this to a `.cpp` file but little harm for now in compiling it a few extra times.